src/state/features/menu/menuSlice.tsx

```diff
@@ -0,0 +1,31 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+// This can also be directly a bool
+interface MenuState {
+  menuOpen: boolean;
+  activeLink: string;
+}
+
+const initialState: MenuState = {
+  menuOpen: false,
+  activeLink: "",
+};
+
+const menuSlice = createSlice({
+  name: "menu",
+  initialState,
+  reducers: {
+    openMenu(state) {
+      state.menuOpen = true;
+    },
+    closeMenu(state) {
+      state.menuOpen = false;
+    },
+    setActiveLink(state, action: PayloadAction<string>) {
+      state.activeLink = action.payload;
+    },
```

Comment:
```suggestion
    },
    handleMenu(state, action) {
      const { payload } = action;
      state.menuOpen = payload.status;
    },
```

Reply:
Love the solution! I end up adding it as below, so I can also add the payloadType

```
reducers: {
    handleMenu(state, action: PayloadAction<boolean>) {
      state.menuOpen = action.payload;
    },
  },
```